pkg/platform/model/vcs.go

```diff
@@ -600,13 +602,37 @@ func CommitInitial(hostPlatform string, langName, langVersion string) (strfmt.UU
 	return res.Payload.CommitID, nil
 }
 
+var versionRe = regexp.MustCompile(`^\d+(\.\d+)+$`)
+
+func isExactVersion(version string) bool {
+	return versionRe.MatchString(version)
+}
+
 func isWildcardVersion(version string) bool {
 	return strings.Index(version, "x") >= 0 || strings.Index(version, "X") >= 0
 }
 
 func VersionStringToConstraints(version string) ([]*mono_models.Constraint, error) {
+	if isExactVersion(version) {
+		return []*mono_models.Constraint{{Comparator: "eq", Version: version}}, nil
+	}
+
 	if !isWildcardVersion(version) {
-		return []*mono_models.Constraint{{Comparator: "eq", Version: version}}, nil // exact version
+		// Ask the Platform to translate a string like ">=1.2,<1.3" into a list of constraints.
+		// Note that:
+		// - The given requirement name does not matter; it is not looked up.
+		// - The given language only needs to exist. It has no bearing on the translation and is not
+		//   present in the requirements read and returned.
+		dummyLanguage := language.Python3.Requirement()
+		changeset, err := reqsimport.Init().Changeset([]byte("name "+version), dummyLanguage)
+		if err != nil {
+			return nil, errs.Wrap(err, "Unable to translate version string into requirement list")
+		}
+		constraints := []*mono_models.Constraint{}
+		for _, change := range changeset {
+			constraints = append(constraints, change.VersionConstraints...)
+		}
+		return constraints, nil
```

Comment:
This sections makes me concerned for cases like `1.2.3rc2` or some oddball like `3.2.1x4`.
```
switch {
case isExactVersion(version):
	// ...
case isWildcardVersion(version):
	// ...
case hasVersionComparator(version):
	// ...
default:
	// the user likely provided a mix of wildcard and version comparator forms - input error
}
```
```
func isExactVersion(v string) bool {
	return !isWildcardVersion(v) && !hasVersionComparator(v)
}

func hasVersionComparator(v string) bool {
	return strings.ContainsAny(v, ">=<,")
}

func isWildcardVersion(v string) bool {
	return strings.HasSuffix(v, ".x") || strings.Contains(v, ".x.")
}
```

Reply:
I have never seen a platform version that is not of the form x.y.z. Also, the ACs said to not perform any sort of processing/interpretation of the version string given. I only have the `isExactVersion()` there to avoid a platform call for performance reasons. I am perfectly content having the platform do all the work in interpreting a version, and erroring out if it is invalid.

Reply:
I think I see what you're saying. My thought is that isExactVersion could be a bit more accurate to its name. @Naatan, is it inappropriate to consider values like 1.2.3rc4 in this case?

Reply:
I'm not sure. @swagile1 what's the contract on ingredient versions?

For that matter; could we have versions that have `.x` in them?

Reply:
We have Python versions of 2.7.17.10, for example. It seems unlikely we'll have unstable versions generally recommended in the foreseeable future though, so I suggest we stick with dot-separated integers for now and worry about names attached to release numbers later.

For what it's worth, `parseversion` from [langtools](https://github.com/ActiveState/langtools/) is probably the closest we have to a standard or contract on ingredient versions.

Reply:
@daved @mitchell-as so based on Shane's response I think the current changes are fine, no need to be more smarter /s.